cmd/tool/parser/parser.go

```diff
@@ -0,0 +1,55 @@
+// Package parser package is responsible for parsing the output of the `go test`
+// command and returning additional info about failred test cases, such as file
+// and line number of failed test.
+package parser
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"gotest.tools/gotestsum/internal/log"
+)
+
+// ParseFailure parses the output of the `go test` for a test failure  and
+// returns the file and line number of the failed test case.
+func ParseFailure(output string) (file string, line int, err error) {
+	scanner := bufio.NewScanner(strings.NewReader(output))
+	for scanner.Scan() {
+		outputLine := scanner.Text()
+		// Usually the failure would contain a line like this:
+		// Error Trace:	/Users/user/proje/path/to/package/some_test.go:42
+		// where the full path to the file is in the same line as "Error Trace:"
+		if strings.Contains(outputLine, "Error Trace") {
```

Comment:
I don't think this I've ever see an "Error Trace" line like this before. Is this from a specific test helper library?

Most test failures I see look like this:

```
fails_test.go:50: failed sub
```

I don't think we can rely on this prefix.